src/encoder.rs

```diff
@@ -24,140 +21,150 @@ use byteorder::{BigEndian, ByteOrder};
 /// );
 /// assert!(encoder::encode(&packet).is_ok())
 /// ```
-pub fn encode(packet: &OscPacket) -> Result<Vec<u8>> {
+pub fn encode(packet: &OscPacket) -> crate::types::Result<Vec<u8>> {
+    let mut bytes = Vec::new();
+    encode_into(packet, &mut bytes)
+        .expect("Failed to write encoded packet into Vec");
```

Comment:
What do you think of returning the error from `encode_into` here?  This would be consistent with the previous behavior and how `encode_into` works.

Reply:
There's a type incompatibility. `encode` has to return the OSC error type, which isn't compatible with `<Vec<u8> as Output>::Err`, which can either be `core::convert::Infallible` (no `std`) or `std::io::Error` (`std`).

It might be worth making a comment noting it, but neither of these can actually happen in this case. `Infallible` by nature, and the `Write` implementation for `Vec<u8>` will only ever return `Ok`.

Reply:
Got it, thank you for the explanation.  In this case it's fine to panic if it should ever error out, which might only happen if there's no more memory to allocate I guess.